src/js/media-controller.js

```diff
@@ -128,6 +169,41 @@ class MediaController extends MediaContainer {
       this.removeAttribute(Attributes.FULLSCREEN_ELEMENT);
     }
     this.#fullscreenElement = element;
+    // Use the getter in case the fullscreen element was reset to "`this`"
+    this.#mediaStore.dispatch({ type: 'fullscreenelementchangerequest', detail: this.fullscreenElement });
```

Comment:
this is a long one 😅 `fullscreenelementchangerequest` 
I would remove request here, `request` is implied I think in each action / dispatch. it could fail...

Reply:
This one's a bit more flexible compared to the prior event rename suggestion. In our current architecture, we wouldn't typically expect a developer to dispatch these events, even in the web components world. i.e. this is more "under the hood" eventing/dispatching. On the plus side, the naming convention is consistent and explicit/clear on what it does with its current name. On the minus side, it's a big boi. My primary concern with `'fullscreenelementchange'` is it sounds like it is an event announcing that `fullscreenElement` **_has_** changed (like other "change" events), when it's actually an event asking to **_cause_** a change. @heff thoughts?

Reply:
A followup note from an IRL conversation with @cjpillsbury: 

`requests` are like Actions in redux. Actions work like events, but are clearly not events because they're Actions. Without the term 'request', and because we're using actual events for this purpose, it's not clear requests are different from other events. It might be clear when you're using the store/provider directly, but not in other contexts.

The store is built in a way where:
1. A change to state is requested
2. That change is calculated
3. The state is updated and propagated

Because of that, it makes sense to include state owner changes as part of that workflow, rather than making a special case for the and potentially adding more complexity to the internal workflow.

So then all that to say, I don't have a good answer here if the goal is to make that event name shorter. I agree it's a big one. Part of me doubts that anyone besides us will ever do anything with this specific event. I know it doesn't mean this topic won't come up again. My gut is to kick the can on this one and leave it as is.